frontend/src/components/advanced-auto-filter/advanced-auto-filter.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+
+import { AutoRiaOption } from '@autoline/shared/common/types/types';
+import { FiltersNames } from '@common/enums/cars/filters-names.enum';
+import { pricesRange, raceRange, yearsRange } from '@common/enums/cars/ranges';
+import { AutocompleteValueType } from '@common/types/cars/autocomplete.type';
+import { BrandDetailsType } from '@common/types/cars/brand-details.type';
+import { CheckboxListDataType } from '@common/types/cars/checkbox-list-data.type';
+import { RangeValueType } from '@common/types/cars/range-item.type';
+import { BrandDetails } from '@components/advanced-auto-filter/brand-details/brand-details';
+import { AutocompleteInput } from '@components/common/autocomplete-input/autocomplete-input';
+import { CheckboxList } from '@components/common/checkbox-list/checkbox-list';
+import { RangeSelector } from '@components/common/range-selector/range-selector';
+import { filtersToQuery } from '@helpers/filters-to-query';
+import { useAppDispatch, useAppSelector } from '@hooks/hooks';
+import UTurnRightIcon from '@mui/icons-material/UTurnRight';
+import { resetAllFilters, setValue } from '@store/car-filter/slice';
+import {
+  useGetFilteredCarsQuery,
+  useGetUsedOptionsQuery,
+} from '@store/queries/cars';
+
+import styles from './styles.module.scss';
+
+const AdvancedAutoFilter: FC = () => {
+  const dispatch = useAppDispatch();
+
+  const filters = useAppSelector((state) => state.carFilter);
+
+  const [queryParams, setQueryParams] = useState<string[][]>();
+
+  const initialBrandDetails = {
+    id: Date.now().toString(),
+    brandId: '',
+    modelId: '',
+  };
+  const [brandDetailsList, setBrandDetailsList] = useState<BrandDetailsType[]>([
+    initialBrandDetails,
+  ]);
+
+  const { data: options, isLoading } = useGetUsedOptionsQuery();
+  useEffect(() => {
+    setQueryParams(filtersToQuery(filters));
+  }, [filters]);
+
+  const { data: filteredCars } = useGetFilteredCarsQuery(queryParams, {
+    skip: !queryParams,
+  });
+
+  // eslint-disable-next-line no-console
+  console.log(filteredCars);
+
+  const handleAddNewDetails = (): void => {
+    setBrandDetailsList([
+      ...brandDetailsList,
+      {
+        id: Date.now().toString(),
+        brandId: '',
+        modelId: '',
+      },
+    ]);
+  };
+
+  const handleBrandDetailsChange = (data: {
+    id: string;
+    brandId: string;
+    modelId: string;
+  }): void => {
+    const { brandId, modelId } = data;
+    setBrandDetailsList([
+      ...brandDetailsList.filter((item) => item.id !== data.id),
+      {
+        id: data.id,
+        brandId,
+        modelId,
+      },
+    ]);
+    if (!filters.brandId.includes(brandId) && brandId !== '') {
+      dispatch(
+        setValue({
+          filterName: FiltersNames.BRAND_ID,
+          value: [...filters.brandId, data.brandId],
+        }),
+      );
+    }
+    if (!filters.modelId.includes(modelId) && modelId !== '')
```

Comment:
`brandId` is named in singular form, but contains an array :eyes:

Reply:
That's because I pass query params to RTK as arras of arrays
![image](https://user-images.githubusercontent.com/49813216/186520867-c5c135d2-d0a7-4d4b-ab28-ec43e219ca01.png)